### PR TITLE
[Coop] ves_icall_property_info_get_default_value.

### PR DESCRIPTION
**Pull request by jaykrell**
This includes resolving a somewhat deep problem involving
a void* that is usually to a int/float, but can be a MonoObject** or at least MonoString**.
An additional MonoStringHandleOut parameter is passed around to address this.
Perhaps it should be MonoObjectHandleOut for more generality.

### CODE REVIEW

**Comment by jaykrell** [comment]
Unrelated failures seen in several PRs.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/metadata/appdomain.c [hunk 1]
```diff
@@ -130,9 +130,13 @@ mono_domain_asmctx_from_path (const char *fname, MonoAssembly *requesting_assemb
 static void
 add_assemblies_to_domain (MonoDomain *domain, MonoAssembly *ass, GHashTable *hash);
 
+#if ENABLE_NETCORE
+
```
Comment (vargaz) [inline mono/metadata/appdomain.c]: Unrelated.

File: mono/metadata/icall-def-netcore.h [hunk 2]
```diff
@@ -226,7 +226,7 @@ ICALL_TYPE(TYPEB, "System.Reflection.Emit.TypeBuilder", TYPEB_1)
 HANDLES(TYPEB_1, "create_runtime_class", ves_icall_TypeBuilder_create_runtime_class, MonoReflectionType, 1, (MonoReflectionTypeBuilder))
 
 ICALL_TYPE(FIELDI, "System.Reflection.FieldInfo", FILEDI_1)
-HANDLES(FILEDI_1, "get_marshal_info", ves_icall_System_Reflection_FieldInfo_get_marshal_info, MonoReflectionMarshalAsAttribute, 1, (MonoReflectionField))
+ HANDLES(FILEDI_1, "get_marshal_info", ves_icall_System_Reflection_FieldInfo_get_marshal_info, MonoReflectionMarshalAsAttribute, 1, (MonoReflectionField))
```
Comment (vargaz) [inline mono/metadata/icall-def-netcore.h]: Extra whitespace.

File: mono/metadata/appdomain.c (hunk 1 above)
Comment (jaykrell) [reply mono/metadata/appdomain.c]: Moved to https://github.com/mono/mono/pull/16622

Will undo here.

File: mono/metadata/icall-def-netcore.h (hunk 2 above)
Comment (jaykrell) [reply mono/metadata/icall-def-netcore.h]: Accidental, will fix.

File: mono/mini/debugger-agent.c [hunk 3]
```diff
@@ -7900,6 +7900,8 @@ collect_interfaces (MonoClass *klass, GHashTable *ifaces, MonoError *error)
 static ErrorCode
 type_commands_internal (int command, MonoClass *klass, MonoDomain *domain, guint8 *p, guint8 *end, Buffer *buf)
 {
+	HANDLE_FUNCTION_ENTER ();
```
[the remark this answers is not in the document]
Comment (jaykrell) [reply mono/mini/debugger-agent.c]: It'd be good to take https://github.com/mono/mono/pull/16620 first.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by akoeplinger** [comment]
@monojenkins build Windows x64 FullAOT

**Comment by akoeplinger** [comment]
@monojenkins build Windows x64 FullAOT+LLVM

**Comment by vargaz** [comment]
@monojenkins build Windows x64 FullAOT
